File: day05/solution.js
let input = require("fs").readFileSync("./data").toString();

function isNice(part2) {
	if (!part2) {
		return line => /([aeiou].*?){3}/.test(line) && /(.)\1/.test(line) && !(/ab|cd|pq|xy/.test(line));
	} else {
		return line => /(..).*?\1/.test(line) && /(.).\1/.test(line);
	}
}

console.log("Part 1:", input.split("\n").filter(isNice(false)).length);
console.log("Part 2:", input.split("\n").filter(isNice(true)).length);

File: day05/solution.py
import re

input = open("data", "r").read()

def isNice1(string):
	return re.search('([aeiou].*?){3}', string) and re.search('(.)\\1', string) and not re.search('ab|cd|pq|xy', string);

def isNice2(string):
	return re.search('(..).*?\\1', string) and re.search('(.).\\1', string);

print "Nice Strings (Part 1):", len(filter(isNice1, input.split('\n')))
print "Nice Strings (Part 2):", len(filter(isNice2, input.split('\n')))
